### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by MarcoAurelioAntonio**
### Project requirements
First of all - make sure that you keep state immutable in reducers!
Secondly, change the default state in your books reducer from an empty array to an array with a few books. One of the goals is to display them in your React components.
Import Redux Provider and your store in the main component of your React app.
Connect your app to the Redux store.
Make sure that list of books displays books from Redux store.
Use useSelector to consume the state.
In your React component responsible for adding new books, add functionality for "Add new" button click:
Import Redux useDispatch.
Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
Takes the inputs from the state and generate a unique id and add them to an object.
Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).
In your React component responsible for removing books, add a functionality for "Remove" button click:
Import Redux useDispatch.
Dispatch a corresponding action (import Action Creator that you created before and use it here).
In your React component responsible for categories, add a functionality for "Check status" button click:
Import Redux useDispatch.
Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

File: src/components/Remove.js [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+
+  const remBook = (e) => {
+    const { id } = e.target;
+    dispatch(removeBook({ id }));
+  }; */
+
   return (
-    <div>Remove</div>
+    <div>
+      {/* <button type="button" id={book.id} onClick={remBook}>Remove</button> */}
```
Comment (t-yanick) [inline src/components/Remove.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

File: src/components/Remove.js [hunk 2]
```diff
@@ -1,7 +1,18 @@
 import React from 'react';
+/* import { useDispatch } from 'react-redux';
+import { removeBook } from '../redux/books/book'; */
```
Comment (t-yanick) [inline src/components/Remove.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

File: src/components/Books.js [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
-      <Book author="Pepito" title="The lodr of los anillos" value={25} category="Action" />
+
+      { books?.map((book) => <Book key={book.id} book={book} value={17} />)}
+
+      {/* <Book author="Pepito" title="The lodr of los anillos" value={25} category="Action" />
       <Book author="Juanito" title="meditacions" value={53} category="Science Fiction" />
-      <Book author="Ranita de Metan" title="enquiridion" value={69} category="Terror" />
-      <Book author="Pinocho" title="blancanieves" value={9} category="Suspenso" />
+      <Book author="Ranita de Metan" title="enquiridion" value={69} category="Terror" /> */}
+      {/* <Book author="Pinocho" title="blancanieves" value={9} category="Suspenso" /> */}
```
Comment (t-yanick) [inline src/components/Books.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @MarcoAurelioAntonio , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

 #### I will leave the comments here, since your `Readme file` is not accessible for this `PR` as you did not make any change on it ####

- [ ] Please kindly update your `readme file` to reflect the changes for this milestone as this will make your `readme file` more informative and professional. Checkout this [template](https://github.com/microverseinc/readme-template) and adapt it to your project. Replace all the information in the various sections of the template with information pertaining to your project.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @MarcoAurelioAntonio 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**